### PR TITLE
Add classes for plpython and postgresql-docs 

### PR DESCRIPTION
**Pull request by deVIAntCoDE**
Hi, i have just added these features:
1. support for plpython package management, 
2. suport for postgresql-docs management. 
My need was for Rehat (Centos 6), as such i haven't investigated implications for Debian and other providers. 
Kindly review and let me know what you think. 
Thanks

### CODE REVIEW

**Comment by igalic** [comment]
should we maybe follow the example of https://github.com/puppetlabs/puppetlabs-postgresql/blob/master/spec/unit/classes/server/plperl_spec.rb and add an rspec test here?

File: manifests/globals.pp [hunk 1]
```diff
@@ -6,8 +6,10 @@
   $contrib_package_name = undef,
   $devel_package_name   = undef,
   $java_package_name    = undef,
+  $docs_package_name    = undef,
   $perl_package_name    = undef,
   $plperl_package_name  = undef,
+  $plpython_package_name = undef,
```
Comment (igalic) [inline manifests/globals.pp]: this pr looks good, except for this line, which irks me :P

**Comment by deVIAntCoDE** [comment]
Adding tests promptly

**Comment by igalic** [comment]
@deVIAntCoDE would you mind [squashing](http://www.git-scm.com/book/en/Git-Tools-Rewriting-History#Squashing-Commits) those commits down to one?
***
also note that when you commit stuff to git and then push it to github, github will only associate commits with your account that it knows. you can either add these commits' email address to your github profile, or git commit --amend the commits' author

**Comment by deVIAntCoDE** [comment]
Hi @igalic,  you mean squash just the last two or all commits in the PR into one?

**Comment by igalic** [comment]
looks gooder. :+1:

File: README.md [hunk 2]
```diff
@@ -289,12 +291,18 @@ This setting can be used to override the default postgresql devel package name.
...
+####`docs_package_name`
+This setting can be used to override the default postgresql docs package name. If not specified, the module will use whatever package name is the default for your OS distro.
+
 ####`perl_package_name`
 This setting can be used to override the default postgresql Perl package name. If not specified, the module will use whatever package name is the default for your OS distro.
 
 ####`plperl_package_name`
 This setting can be used to override the default postgresql PL/perl package name. If not specified, the module will use whatever package name is the default for your OS distro.
 
+####`plpython_package_name`
```
Comment (cyberious) [inline README.md]: We have duplicate readme below

File: README.md [hunk 3]
```diff
@@ -393,6 +401,9 @@ Value to pass through to the `package` resource when creating the server instanc
 ####`plperl_package_name`
 This sets the default package name for the PL/Perl extension. Defaults to utilising the operating system default.
 
+####`plpython_package_name`
```
Comment (cyberious) [inline README.md]: Above is probably a better description

File: manifests/globals.pp [hunk 4]
```diff
@@ -6,8 +6,10 @@
   $contrib_package_name = undef,
   $devel_package_name   = undef,
   $java_package_name    = undef,
+  $docs_package_name    = undef,
   $perl_package_name    = undef,
   $plperl_package_name  = undef,
+  $plpython_package_name= undef,
```
Comment (cyberious) [inline manifests/globals.pp]: Please add spaces otherwise will fail lint

File: manifests/params.pp [hunk 5]
```diff
@@ -34,7 +34,9 @@
         $contrib_package_name = pick($contrib_package_name,'postgresql-contrib')
         $devel_package_name   = pick($devel_package_name, 'postgresql-devel')
         $java_package_name    = pick($java_package_name, 'postgresql-jdbc')
+        $docs_package_name    = pick($docs_package_name, 'postgresql-docs')
         $plperl_package_name  = pick($plperl_package_name, 'postgresql-plperl')
+        $plpython_package_name= pick($plpython_package_name, 'postgresql-plpython')
```
Comment (cyberious) [inline manifests/params.pp]: Spaces for lint.

File: manifests/globals.pp (hunk 4 above)
Comment (deVIAntCoDE) [inline manifests/globals.pp]: i removed the spaces as per a previous review request to maintain the alignment. I could put the spaces back and/or rename 'plpython' to say 'plpythn' to maintain the alignment, which would you prefer?

File: manifests/globals.pp [hunk 6]
```diff
@@ -6,8 +6,10 @@
   $contrib_package_name = undef,
   $devel_package_name   = undef,
   $java_package_name    = undef,
+  $docs_package_name    = undef,
   $perl_package_name    = undef,
   $plperl_package_name  = undef,
+  $plpython_pkg_name    = undef,
```
Comment (mhaskel) [inline manifests/globals.pp]: This should be `$plpython_package_name` for consistency with the other parameters. In order to maintain `=` alignment you need to add spaces to all of the other parameters. This makes the change bigger but is necessary for consistency.

File: manifests/lib/docs.pp [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# This class installs the postgresql-docs See README.md for more
+# details.
+class postgresql::lib::docs (
+  $package_name   = $postgresql::params::docs_package_name,
+  $package_ensure = 'present'
```
Comment (mhaskel) [inline manifests/lib/docs.pp]: Please use a trailing comma.

File: manifests/params.pp [hunk 8]
```diff
@@ -34,7 +34,9 @@
         $contrib_package_name = pick($contrib_package_name,'postgresql-contrib')
         $devel_package_name   = pick($devel_package_name, 'postgresql-devel')
         $java_package_name    = pick($java_package_name, 'postgresql-jdbc')
+        $docs_package_name    = pick($docs_package_name, 'postgresql-docs')
         $plperl_package_name  = pick($plperl_package_name, 'postgresql-plperl')
+        $plpython_pkg_name    = pick($plpython_pkg_name, 'postgresql-plpython')
```
Comment (mhaskel) [inline manifests/params.pp]: This should be `$plpython_package_name`. You will need to add spaces to all other parameter lines to keep alignment.

File: manifests/params.pp [hunk 9]
```diff
@@ -48,7 +50,9 @@
         $contrib_package_name = pick($contrib_package_name,"postgresql${package_version}-contrib")
         $devel_package_name   = pick($devel_package_name, "postgresql${package_version}-devel")
         $java_package_name    = pick($java_package_name, "postgresql${package_version}-jdbc")
+        $docs_package_name    = pick($docs_package_name, "postgresql${package_version}-docs")
         $plperl_package_name  = pick($plperl_package_name, "postgresql${package_version}-plperl")
+        $plpython_pkg_name    = pick($plpython_pkg_name, "postgresql${package_version}-plpython")
```
Comment (mhaskel) [inline manifests/params.pp]: and `$plpython_package_name`

File: manifests/params.pp [hunk 10]
```diff
@@ -91,6 +95,7 @@
       $contrib_package_name = pick($contrib_package_name,'undef')
       # Archlinux postgresql package provides plperl
       $plperl_package_name  = pick($plperl_package_name, 'undef')
+      $plpython_pkg_name    = pick($plpython_pkg_name, 'undef')
```
Comment (mhaskel) [inline manifests/params.pp]: and `$plpython_package_name`.

File: manifests/params.pp [hunk 11]
```diff
@@ -135,6 +140,7 @@
       $java_package_name    = pick($java_package_name, 'libpostgresql-jdbc-java')
       $perl_package_name    = pick($perl_package_name, 'libdbd-pg-perl')
       $plperl_package_name  = pick($plperl_package_name, "postgresql-plperl-${version}")
+      $plpython_pkg_name    = pick($plpython_pkg_name, "postgresql-plpython-${version}")
```
Comment (mhaskel) [inline manifests/params.pp]: and `$plpython_package_name`

File: manifests/server.pp [hunk 12]
```diff
@@ -7,6 +7,7 @@
   $package_ensure             = $postgresql::params::package_ensure,
 
   $plperl_package_name        = $postgresql::params::plperl_package_name,
+  $plpython_pkg_name          = $postgresql::params::plpython_pkg_name,
```
Comment (mhaskel) [inline manifests/server.pp]: `$plpython_package_name`.

File: manifests/server/plpython.pp [hunk 13]
```diff
@@ -0,0 +1,19 @@
+# This class installs the PL/Python procedural language for postgresql. See
+# README.md for more details.
+class postgresql::server::plpython(
+  $package_ensure = 'present',
+  $package_name   = $postgresql::server::plpython_pkg_name
```
Comment (mhaskel) [inline manifests/server/plpython.pp]: Please use a trailing comma

**Comment by mhaskel** [comment]
@deVIAntCoDE I made a number of inline comments. This should also be squashed down to one commit.

**Comment by DavidS** [comment]
I've rebased and squashed this into #634

**Pull request closed without merging**